docs/docs/concepts/Cluster_Support.md

```diff
@@ -0,0 +1,108 @@
+
+* `async_client.runOnShards` - run the remote function on all the shards (including the current shard). Returns a promise that once resolve will give 2 nested arrays, the first contains another array with the results from all the shards and the other contains an array of errors (`[[res1, res2, ...],[err1, err2, ..]]`)
+* `async_client.runOnKey` - run the remote function on the shard responsible for a given key. Returns a promise that once resolve will give the result from the remote function execution or raise an exception in case of an error.
+
+The following example register a function that will return the total amount of keys on the cluster. The function will use the remote function define above:
+
+```js
+redis.registerAsyncFunction("my_dbsize", async(async_client) => {
+    let res = await async_client.runOnShards("dbsize");
+    let results = res[0];
+    let errors = res[1];
+    if (errors.length > 0) {
+        return errors;
+    }
+    let sum = BigInt(0);
+    results.forEach((element) => sum+=BigInt(element));
+    return sum;
+});
+```
+
+First the function executes a remote function on all shards that returns the amount of keys on each shard, then the function summarize all the results and returns it.
+
+The full code will look like this:
+
+```js
+#!js name=lib api_version=1.0
+
+redis.registerClusterFunction("dbsize", async(client) => {
+    return client.block((client) => {
+        return client.call("dbsize").toString();
+    });
+});
+
+redis.registerAsyncFunction("test", async(async_client) => {
+    let res = await async_client.runOnShards("dbsize");
+    let results = res[0];
+    let errors = res[1];
+    if (errors.length > 0) {
+        return errors;
+    }
+    let sum = BigInt(0);
+    results.forEach((element) => sum+=BigInt(element));
+    return sum;
+});
+```
+
+## Arguments and Results Serialization
+
+It is possible to pass arguments to the remote function. The arguments will be given to the remote function after the `async_client`. The following example shows how to get a value of a key from any shard in the cluster:
+
+```js
+#!js name=lib api_version=1.0
+const remote_get = "remote_get";
+
+redis.registerClusterFunction(remote_get, async(client, key) => {
+    let res = client.block((client) => {
+        return client.call("get", key);
+    });
+    return res;
+});
+
+redis.registerAsyncFunction("test", async (async_client) => {
+    return await async_client.runOnKey("x", remote_get, "x");
+});
+```
+
+The function `test` will return the value of `x` regardless on which shard the function was executed on.
+
+The remote function arguments and results are serialized in the following way:
+
+1. If the argument is of `ArrayBuffer` type, the data will be sent as is.
+2. Otherwise, RedisGears will try to serialize the give arguments (or the return value) as json using `JSON.stringify`. A serialization failure will cause an error to be raised.
```

Comment:
```suggestion
2. Otherwise, RedisGears will try to serialize the give arguments (or the return value) as JSON using `JSON.stringify`. A serialization failure will cause an error to be raised.
```

Reply:
Not sure if the use of RedisGears here is proper or not. Deferring to PM.